src/ripple/app/tx/impl/SetTrust.cpp

```diff
@@ -133,14 +133,22 @@ SetTrust::preclaim(PreclaimContext const& ctx)
     auto const sleDst = ctx.view.read(keylet::account(uDstAccountID));
 
     // If the destination has opted to disallow incoming trustlines
-    // then honour that flag
+    // then honour that flag ONLY if the trustline doesn't exist
     if (ctx.view.rules().enabled(featureDisallowIncoming))
     {
         if (!sleDst)
             return tecNO_DST;
 
         if (sleDst->getFlags() & lsfDisallowIncomingTrustline)
-            return tecNO_PERMISSION;
+        {
+            if (ctx.view.rules().enabled(fixDisallowIncomingV1) &&
```

Comment:
If I were reading this code in the future, I think I'd find the changed comment on line 136 misplaced.  My inclination is to remove the change from line 136 and, instead, add something like this comment just in front of this `if`:
```
            // The original implementation of featureDisallowIncoming was
            // too restrictive.  If
            //   o fixDisallowIncomingV1 is enabled and
            //   o The trust line already exists
            // Then allow the TrustSet.
```
I think this is an improvement.  But it's only a suggestion.  Take it, modify it, or leave it as you will.